### PR TITLE
Add multi crystals to ispyb comment

### LINKED ISSUE #476: Handle zocalo returning multiple crystals
In some samples there could be more than one crystal. In this case we're prototyping Zocalo returning a list of centres, with the strength of each one. The json returned from zocalo will look something like:
```
[
  {
    "centre_of_mass": [
      4.636612650133922,
      6.038044420033806,
      2.612136870930547
    ],
    "max_count": 156032.0,
    "max_voxel": [
      3,
      7,
      1
    ],
    "n_voxels": 65,
    "total_count": 6671044.0
  },
  {
    "centre_of_mass": [
      3.5,
      11.5,
      11.039976728723405
    ],
    "max_count": 103952.0,
    "max_voxel": [
      3,
      11,
      11
    ],
    "n_voxels": 2,
    "total_count": 192512.0
  }
]
```

## Acceptance Criteria
* Artemis chooses which crystal to go to based on the most `total_counts`
* System tests and fake_zocalo updated to reflect this

### LINKED ISSUE #513: Add multi crystals to ispyb comment
We are now getting multiple crystals back from zocalo. We should put in the ispyb comment something like:
```
Strength	Position (x,y,z)	Size (x,y,z)
192		192, 30, -35		100, 20, 20
```
to show this data. In the long term it would be good to get this as into ispyb as a proper table but requires discussion with at least @KarlLevik.

### PR DESCRIPTION
**Pull request by noemifrisina**
Fixes #513 
Fixes #476 

### To test:
1. Check  `test_zocalo_system` doesn't fail
2. Run all tests

### CODE REVIEW

File: fake_zocalo/__main__.py [hunk 1]
```diff
@@ -72,6 +77,22 @@ def main():
         "recipe-pointer": 1,
     }
 
+    multi_crystal_result = {
```
Comment (DominicOram) [inline fake_zocalo/__main__.py:80]: Nit: This is getting a bit messy with these 3 basically the same maybe something like:
```python
def make_result(payload):
    return {
        "environment": {"ID": "6261b482-bef2-49f5-8699-eb274cd3b92e"},
        "payload": payload,
        "recipe": {
            "start": [[1, payload]],
            "1": {
                "service": "Send XRC results to GDA",
                "queue": "xrc.i03",
                "exchange": "results",
                "parameters": {"dcid": "2", "dcgid": "4"},
            },
        },
        "recipe-path": [],
        "recipe-pointer": 1,
    }

results = defaultdict(lambda : make_result([TEST_RESULT_LARGE]), {"NO_DIFF": make_result([]), "MULTI_X": make_result([*TEST_RESULT_LARGE, *TEST_RESULT_SMALL]})

...

result = results[prefix]
```

File: src/artemis/external_interaction/callbacks/fgs/zocalo_callback.py [hunk 2]
```diff
@@ -88,6 +88,34 @@ def wait_for_results(self, fallback_xyz: Point3D) -> Point3D:
...
+            # Sort from strongest to weakest in case of multiple crystals
+            raw_results = sorted(
+                raw_results, key=lambda d: d["total_count"], reverse=True
+            )
+            LOGGER.info(f"Zocalo: found {len(raw_results)} crystals.")
+            multi_crystal_msg = (
+                f"Found multiple crystals: {len(raw_results)}."
+                if len(raw_results) > 1
+                else ""
+            )
```
Comment (DominicOram) [inline src/artemis/external_interaction/callbacks/fgs/zocalo_callback.py:100]: Nit: I would just remove this message, it should be obvious from the rest of the comment

File: src/artemis/external_interaction/callbacks/fgs/zocalo_callback.py [hunk 3]
```diff
@@ -88,6 +88,34 @@ def wait_for_results(self, fallback_xyz: Point3D) -> Point3D:
...
+                            operator.sub, res["bounding_box"][1], res["bounding_box"][0]
+                        )
+                    )
+                )
+                multi_crystal_msg += (
+                    f"Crystal {n+1}: "
+                    f"Strength {res['total_count']} "
+                    f"Position (x,y,z) {res['centre_of_mass']} "
+                    f"Size (x,y,z) {bboxes[n]} "
+                )
```
Comment (DominicOram) [inline src/artemis/external_interaction/callbacks/fgs/zocalo_callback.py:116]: Should: This is potentially going to be hard to read, can we put in a carriage return between each crystal?

**Review by DominicOram — CHANGES_REQUESTED** [review]
Great, thank you! Good test coverage. Tests pass and looks good, minor comments in code.

File: src/artemis/external_interaction/callbacks/fgs/zocalo_callback.py (hunk 3 above)
Comment (noemifrisina) [reply src/artemis/external_interaction/callbacks/fgs/zocalo_callback.py]: Done, good point

**Review by DominicOram — APPROVED** [review]
Tests pass, code looks good, thank you!

**Pull request merged**